src/github.com/getlantern/lantern-android/client/client_test.go

```diff
@@ -0,0 +1,108 @@
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+)
+
+const listenProxyAddr = "127.0.0.1:9997"
+
+var testURLs = map[string][]byte{
+	"http://www.google.com/humans.txt": []byte("Google is built by a large team of engineers, designers, researchers, robots, and others in many different sites across the globe. It is updated continuously, and built with more tools and technologies than we can shake a stick at. If you'd like to help us out, see google.com/careers.\n"),
+	// TODO: This is not working, we actually need to implement a CONNECT request to proxy HTTPs traffic.
+	// "https://www.google.com/humans.txt": []byte("Google is built by a large team of engineers, designers, researchers, robots, and others in many different sites across the globe. It is updated continuously, and built with more tools and technologies than we can shake a stick at. If you'd like to help us out, see google.com/careers.\n"),
```

Comment:
Hi @xiam, since @oxtoacart is on vacation, I'll chime in, but I'll let him correct me if I happen to be mistaken. You'll want to configure http.Transport with a Proxy similar to 
```go
	client := &http.Client{
		Transport: &http.Transport{
			Proxy: func(req *http.Request) (*url.URL, error) {
				return url.Parse(listenProxyAddr)
			},
			Dial: func(n, a string) (net.Conn, error) {
				return net.Dial("tcp", listenProxyAddr)
			},
		},
	}
```
A similar example can be found in flashlight_test.go